testsuite/tests/statmemprof/comballoc.ml

```diff
@@ -0,0 +1,35 @@
+(* TEST
+   flags = "-g"
+   compare_programs = "false" *)
+
+open Gc.Memprof
+
+let f4 n = (n,n,n,n)
+
+let[@inline never] f n =
+  (n, (n, n, f4 n))
+
+let test sampling_rate =
+  let counts = Array.make 257 0 in
+  start ~callstack_size:10
+    ~minor_alloc_callback:(fun info ->
+      counts.(info.size) <- counts.(info.size) + info.n_samples; Some info.size)
+    ~minor_dealloc_callback:(fun sz ->
+      assert (sz = 2 || sz = 3))
```

Comment:
Using a deallocation callback without a promotion callback is weird. Could you add a promotion callback to make sure this assertion is checked even on promoted blocks?

Reply:
Done. (It's not quite this assertion, though: I expect it to be false on promoted blocks).